routing/router.go

```diff
@@ -1596,6 +1596,10 @@ type LightningPayment struct {
 	// hop. If nil, any channel may be used.
 	OutgoingChannelID *uint64
 
+	// LastHop is the pubkey of the last node before the final destination
+	// is reached. If nil, any node may be used.
+	LastHop *route.Vertex
```

Comment:
Do we care about validating whether this is an existing node in the graph? Path finding would result in `unable to find a path to destination`, which can be ambiguous.

Reply:
We also don't do this for the outgoing channel restriction. As this is advanced functionality, I am ok with leaving it up to the caller to make sure this doesn't happen. What do you think?

Maybe if we start fixing https://github.com/lightningnetwork/lnd/issues/1680, we can add two new payment failures:
`no_outgoing_channel` (with enough balance and not filtered out) and `no_last_hop` (with enough capacity and not filtered out).

Reply:
We can defer it until then.